microprofile/WFLY-17137-microprofile-jakarta-ee-10.adoc

```diff
@@ -0,0 +1,185 @@
+= MicroProfile aligned with Jakarta EE 10
+:author:            Kabir Khan
+:email:             kkhan@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+This issue is about upgrading to most parts of MicroProfile Platform 6, which is aligned with Jakarta EE 10. 'most' is used since a decision has been made to not include the new MicroProfile Metrics version, as the inclusion of MicroMeter tracked by https://issues.redhat.com/browse/EAP7-1686[EAP7-1686] provides our observability strategy.
+
+The below table shows the versions of each specification in MicroProfile Platform 5, the current versions included in WildFLy, and the planned version for MicroProfile Platform 6. Only the minor versions are included, we will use the latest micro versions of each at the time of release. It also contains a column indicating whether something changes the functionality provided by a specification.
+
+[cols="1,1,1,1,1"]
+|===
+|Spec | MP 5 Version | Current WildFly Version | MP 6 Version | Updated Functionality (y/n)
+
+|Config
+|3.0
+|3.0
+|3.0
+|n
+
+|Fault Tolerance
+|4.0
+|4.0
+|4.0
+|n
+
+|Health
+|4.0
+|4.0
+|4.0
+|n
+
+|JWT
+|2.0
+|2.0
+|TODO: 2.0 (might become 2.1)
+|TODO: ??? (depends on 2.1)
+
+|+++<s>Metrics</s>+++ +
+Will not include
+|+++<s>4.0</s>+++
+|+++<s>4.0</s>+++
+|+++<s>5.0</s>+++
+|+++<s>y</s>+++
+
+|OpenAPI
+|3.0
+|3.0
+|3.1
+|TODO: ??
+
+|OpenTracing +
+Dropped in MP 6
+|3.0
+|3.0
+|-
+|-
+
+|Rest Client
+|3.0
+|3.0
+|3.0
+|n
+
+|Telemetry +
+New in MP 6
+|-
+|-
+|1.0
+|y -
+
+|===
+
+TODO: More about MP Telemetry - will we support it? If we are not that needs to be pointed out. If we are we need links to its RFE.
+
+
+== Issue Metadata
+
+* https://issues.redhat.com/browse/WFLY-17137[WFLY-17137] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+* https://issues.redhat.com/browse/EAP7-1861[EAP7-1861] - Provide Support for MicroProfile.Next version that Aligns with Jakarta EE 10
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-17138[WFLY-17138] - Convert MP Metrics subsystem to Legacy Mode
```

Comment:
One more thing that comes to mind is how to handle the base metrics subsystem if we drop MP metrics. Should it have similar capabilities as Micrometer now?

Reply:
This is more a topic for https://github.com/wildfly/wildfly-proposals/pull/445.

Currently the Micrometer subsystem exposes JVM metrics and also WF management API metrics, the latter in a manner very similar to the base metrics subsystem.